### ISSUE TITLE
conda terminates without output or even an error message

### ISSUE THREAD

**Issue by kevglynn** [ContinuumIO/anaconda-issues#7177]
Running any conda commands yields no output, not even an error message. 

Moreover, `which conda` correctly reflects what I have set in my $PATH (i.e. points to anaconda2). 

![screen shot 2017-11-07 at 10 47 04 am](https://user-images.githubusercontent.com/12972584/32522779-ade9f3b8-c3e6-11e7-94f2-81c7af508227.png)

**Comment by jjhelmus** [comment]
I've seem this happen when one of the conda requirements (e.g. requests) accidentally get removed or mangled.  What is the result of `python -m conda info`?

**Comment by kevglynn** [comment]
again, terminates with no output.

how to investigate these requests?

**Comment by jjhelmus** [comment]
Does the Python REPL work in a normal manner? If not something may be amiss with the ncurses or readline libraries.  

Assuming the REPL works if you can investigate conda via its Python interface that might offer some hints that would help get to the bottom of the issue.  The `conda` module can be imported in a python REPL as follows:

```
~$ python
Python 3.6.3 |Anaconda, Inc.| (default, Nov  3 2017, 19:19:16) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import conda
>>> conda.__version__
'4.3.30'
>>> from conda.cli import main
>>> main('info', '--all')
Current conda install:

               platform : linux-64
          conda version : 4.3.30
       conda is private : False
      conda-env version : 4.3.30
    conda-build version : 2.1.17
         python version : 3.6.3.final.0
       requests version : 2.14.2
       root environment : /home/jhelmus/miniconda3  (writable)
    default environment : /home/jhelmus/miniconda3
       envs directories : /home/jhelmus/miniconda3/envs
                          /home/jhelmus/.conda/envs
          package cache : /home/jhelmus/miniconda3/pkgs
                          /home/jhelmus/.conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/main/linux-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : /home/jhelmus/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.30 requests/2.14.2 CPython/3.6.3 Linux/3.18.0-15958-g5ed7bd9927cb debian/stretch/sid glibc/2.23    
                UID:GID : 1000:1000
...
```

Alternatively these can be run from the command line using the `-c` switch:

```
~$ python -c "import conda; print(conda.__version__)"
4.3.30
~$ python -c "from conda.cli import main; main('info', '--all')"                                                                                                  
Current conda install:

               platform : linux-64
          conda version : 4.3.30
       conda is private : False
      conda-env version : 4.3.30
    conda-build version : 2.1.17
         python version : 3.6.3.final.0
...
```

**Comment by kevglynn** [comment]
kev ~ $ python
Python 2.7.14 |Anaconda, Inc.| (default, Oct  5 2017, 02:28:52) 
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import conda
>>> conda.__version__
u'4.3.30'
>>> from conda.cli import main
>>> main('info', '--all')
1
>>> 
KeyboardInterrupt
>>> 
kev ~ $ python -c "import conda; print(conda.__version__)"
4.3.30
kev ~ $ python -c "from conda.cli import main; main('info', '--all')"
kev ~ $ 


still seem to be drawing a blank here... :(

**Comment by jjhelmus** [comment]
Odd, I'm not sure what is wrong.  Do any other conda commands give output?  Some commands to try would be `conda list`, `conda help` and `conda install foobar` (should give an error since foobar does not exist)?

**Comment by ramanpreet9** [comment]
I am having the same issue as @kevglynn 

Did you figure out what the source is?

**Comment by rjqiao** [comment]
same problem..

**Comment by ramanpreet9** [comment]
@jjhelmus typing all of those commands doesnt print any output. 
no error message for conda install foobar either.


Is there any way to completely uninstall conda and reinstall? i tried the normal way rm rf anaconda 3  but i got the same issues after reinstalling.

**Comment by jjhelmus** [comment]
I'm at a lost here.  Without a traceback this is really difficult to debug.  

On macOS conda can be removed completely using `rm -rf` on the directory in question.  You might try installing a few different version of Miniconda rather than Anaconda to see if the issue persists.

**Comment by vsmolyakov** [comment]
I had the same problem. I found that installing an older version of miniconda worked.  
In my case: `>> bash Miniconda2-4.3.11-Linux-x86_64.sh`
which I found at: https://repo.continuum.io/miniconda/

**Comment by ramanpreet9** [comment]
I reinstalled ubuntu and this problem didn't re-occur.  Yep its hard to debug when we don't have any error log.

**Comment by phuicy** [comment]
I have also had this issue on Scientific Linux

**Comment by phuicy** [comment]
I solved this issue by unsetting my existing PYTHONPATH.

**Comment by mica5** [comment]
I'm experiencing all the same issues in this ticket, conda version 4.3.29 ; this does not help me: "I solved this issue by unsetting my existing PYTHONPATH."

I hope these following traceback pieces are helpful.

```
> /Users/mica/anaconda3/lib/python3.5/site-packages/conda/cli/main.py(121)_main()
    119     # REMOVE
    120     for module in modules:
--> 121         imported = importlib.import_module(module)
    122         imported.configure_parser(sub_parsers)
    123         if "update" in module:
ImportError: cannot import name 'InsecureRequestWarning'
```

value of `module` is 'conda.cli.main_update'

```
ImportError: cannot import name 'InsecureRequestWarning'
> /Users/mica/anaconda3/lib/python3.5/importlib/__init__.py(126)import_module()
    124                 break
    125             level += 1
--> 126     return _bootstrap._gcd_import(name[level:], package, level)
    127
    128
```

(the following block is basically the same as the previous one)

```
ImportError: cannot import name 'InsecureRequestWarning'
> /Users/mica/anaconda3/lib/python3.5/importlib/__init__.py(126)import_module()
    124                 break
    125             level += 1
1-> 126     return _bootstrap._gcd_import(name[level:], package, level)
    127
    128
```

in this context:

```
ipdb> name
'conda.cli.main_update'
ipdb> level
0
ipdb> name[level:]
'conda.cli.main_update'
ipdb> package is None
True
ipdb> _bootstrap
<module 'importlib._bootstrap' (frozen)>
ipdb> _bootstrap._gcd_import
<function _gcd_import at 0x101a76048>
```

I can't step into _bootstrap._gcd_import otherwise I would provide more details.

**Comment by mica5** [comment]
in the meantime, is there any way I can update anaconda while this isn't working, without blowing away my site-packages?

**Comment by mica5** [comment]
TLDR I updated requests using `pip install -U requests` and now `conda` is working.

I dug further and found:
~/anaconda3/lib/python3.5/site-packages/conda/core/repodata.py, line 22:
`from requests.packages.urllib3.exceptions import InsecureRequestWarning`
I commented that line out and now `conda` works, but now I'm trying to figure out how to update anaconda. when I tried to install anaconda from https://www.anaconda.com/download/#download it saw that anaconda is already installed and gave me an error message: "Anaconda3 is already installed in /Users/mica/anaconda3. Use 'conda update anaconda3' to update Anaconda3."
I tried running "conda update anaconda3" but that also gave me an error message:
```
PackageNotInstalledError: Package is not installed in prefix.
  prefix: /Users/mica/anaconda3
  package name: anaconda3
```

after some more google searching I decided to upgrade my version of `requests`, which fixed `conda`. I don't know how `conda` got out of sync with `requests` because I don't remember changing `conda`, but I also tend to use ~/anaconda3/bin/pip for installing things.. maybe I should be using `conda install` instead?..

**Comment by csoja** [comment]
At the moment, conda is not very aware of what you install with pip.  You can easily end up in a bad state mixing pip installed packages with conda installed packages unless you are extremely careful and really know what you are doing.  

On a Unix system, you can force a reinstall of Miniconda/Anaconda without blowing away everything. Refer to the documentation for more details: https://docs.anaconda.com/anaconda/user-guide/troubleshooting#error-message-on-miniconda-install-already-installed

**Comment by mica5** [comment]
@csoja thank you for that note! sorry for taking this ticket off track a little, but when I started with anaconda I don't remember reading any warnings about that issue with anaconda+pip.. maybe I missed it? is there a solid piece of documentation/introduction somewhere that includes that and all other warnings and best practices, which all users of anaconda should be familiar with? I see at https://docs.anaconda.com/anaconda/ "You can download other packages using the pip install command that is installed with Anaconda.", but no mention of a warning. I'm imagining something that says "when installing a package, always first try to use `conda install`, and if that fails, try pip".

**Comment by csoja** [comment]
@mica5 Thank you for the suggestion.  We do have plans to improve the current pip/conda experience - but for now, we should do a better job documenting the current recommendation.

**Comment by mica5** [comment]
@csoja thank you! I wonder if @kevglynn experienced the same issue I did and whether updating requests would fix that issue too.. it could even be a a similar issue with a different library than requests in their case. if that is the case, then they might need to do a similar thing like I did with the debugging to find out which library it was, and update it manually with pip..

**Comment by alvinthai** [comment]
@mica5

I'm experiencing all the same issues in this ticket, and ```pip install -U requests``` did not fix the issue

**Comment by mica5** [comment]
I'm sorry I don't have much time so I'm going to try answer quickly

csoja:
> At the moment, conda is not very aware of what you install with pip. You can easily end up in a bad state mixing pip installed packages with conda installed packages unless you are extremely careful and really know what you are doing.

I know I posted that I upgraded requests using pip, but I remember I eventually ended up reinstalling anaconda. ever since then, I'm better about always trying to make conda environments and installing using conda first. when I want to install a new package I always search google "conda install somepackage" - sometimes a webpage on anaconda's website like this shows up: https://anaconda.org/conda-forge/ipython-sql with a -c CHANNEL option set. if I can't find anything like that, that's when I use pip.

**Comment by alvinthai** [comment]
Yeah I ended up reinstalling anaconda as well. Seems to be the only clean way to fix this

**Comment by JavierReyes945** [comment]
I just experienced the same issue (conda commands where giving no output). It turned out to be a problem in the .condarc config file (I was running conda from the cmd in Windows). When opened the Anaconda prompt, it showed an error when loading the .condarc file at line X column Y. Just FYI, it might help others, but also relevant to know why it was not showing any error message when the command was called form cmd.

**Comment by codebraker** [comment]
> I just experienced the same issue (conda commands where giving no output). It turned out to be a problem in the .condarc config file (I was running conda from the cmd in Windows). When opened the Anaconda prompt, it showed an error when loading the .condarc file at line X column Y. Just FYI, it might help others, but also relevant to know why it was not showing any error message when the command was called form cmd.

Perfect solution.... This really worked out for me. There was an error related to YAML syntax in my .condarc file and after fixing it, the conda commands work just right for me.

**Comment by jagadish20** [comment]
There might me some error in .condarc file.

**Comment by JoaquinSiabra** [comment]
I had the same problem. I have installed Keras with pip. Solved with 'conda install keras'.

**Comment by anotheralex** [comment]
I had the same problem. In my case, the solution was much simpler: I recreated the .condarc file. I had created it by copying the content from another file and perhaps I picked up a strange, invisible character or something.

**Comment by srinivaschitturi** [comment]
I faced a similar issue. Trying the above commands of conda info, conda install etc worked in powershell anaconda prompt
